src/fundus/scraping/pipeline.py

```diff
@@ -120,7 +148,7 @@ def article_gen() -> Iterator[Article]:
                     async_next(interleave, None)
                 )
                 batch_time = time.time() - start_time
-                if delay:
+                if isinstance(delay, Delay):  # typeguard
```

Comment:
No, since this is used within a local function `mypy` does not know if this is called after or before `delay` is parsed, the type `mypy` deduces here is `Union[float, Delay, None]`. The `# typeguard` comment is just a small comment that this is about type guarding the parameter. This hole logic will disappear with #271

Reply:
Ok, all good.